### ISSUE TITLE
Calling an extension method from within an extension method defined on the same class

### ISSUE THREAD

**Issue by markyys** [manifold-systems/manifold#167]
I'm very new to using Manifold, and new to java in general, but it seems to me this kind of thing should be possible:

```
MyModule.extensions.java.lang.Object;

@extension
public class Foo {
	public String SomeFunction(@This Object self){
		return "a string";
	}

	public void AnotherFunction(@this Object self){
		return self.SomeFunction();
	}

	public static void main(String[] args){
		"".AnotherFunction();
	}
}
```

but it fails at compile time saying that `The method SomeFunction() is undefined for the type Object`.

Is it not possible to call an extension method from within an extension method for some reason, or am I just doing something wrong here?

I should note that if I call `".SomeFunction()`' directly that works as expected (assuming that I remove `AnotherFunction` entirely, of course).

**Comment by rsmckinney** [comment]
@markyys Hi. This does indeed work -- Manifold supports self referencing extensions.  Maybe you forgot to make the extension methods static? [See the docs for more info](https://github.com/manifold-systems/manifold/tree/master/manifold-deps-parent/manifold-ext#extension-classes-via-extension).

Here's a working example:
```java
package MyModule.extensions.java.lang.Object;

import manifold.ext.api.Extension;
import manifold.ext.api.This;

@Extension
public class MyObjExt {
    public static String SomeFunction(@This Object self) {
        return "a string";
    }

    public static String AnotherFunction(@This Object self) {
        return self.SomeFunction();
    }

    public static void main(String[] args) {
        "".AnotherFunction();
    }
}
```

**Comment by markyys** [comment]
I typed that example from memory; I should have copy and pasted.

Interestingly, this does work:
```
package MyModule.extensions.java.lang.Object;

import manifold.ext.api.Extension;
import manifold.ext.api.This;

@Extension
public class TestObject {
	public static void new_function(@This Object self){
		System.out.println(self.another_function());
	}
	public static String another_function(@This Object self){
		return "new string";
	}
	public static void main(String[] args){
		"test".new_function();
	}
}
```

but if I add another file:
```package MyModule;

public class ExtensionTest {
	public static void main(String[] args){
		ExtensionTest.ObjectTest();
	}

	private static void ObjectTest(){
		"test %s".new_function();
	}
}
```

that fails with the aforementioned error.
If the class is defined in the same package I shouldn't have to do anything special to apply it, right?
Normally I would say `import MyModule.TestObject` but that also has no effect since I'm inside that package.,

Again, this might be a matter of me just not being familiar with basic java principles.

**Comment by markyys** [comment]
Okay, I figured it out. The problem is, indeed, lack of basic java knowledge.

tl;dr: I was not actually recompiling all source files, so things got wonky.
